### PR TITLE
03 1970 Fix Forms don’t open after having opened a visit form.

### PR DESCRIPTION
**Pull request by sainingo**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

[Walkthrough Video](https://www.loom.com/share/d895202dde9541f4a373083386738433)

## Screenshots

startVisitvsist

## Related Issue

https://issues.openmrs.org/projects/O3/issues/O3-1970

### CODE REVIEW

File: packages/esm-patient-chart-app/src/actions-buttons/start-visit.component.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const StartVisitOverflowMenuItem: React.FC<StartVisitOverflowMenuItemProps> = ({
           }}
         >
           <span className="cds--overflow-menu-options__option-content">
-            {!startVisitLabel ? <>{t('startVisit', 'Start visit')}</> : startVisitLabel}
+            {startVisitLabel ? <>{t('startVisit', 'Start visit')}</> : startVisitLabel}
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/actions-buttons/start-visit.component.tsx:29]: Doesn't this do things incorrectly, i.e., if a user set the `startVisitLabel` via config, they get the translated "Start visit" screen, whereas if they did not, they get the `startVisitLabel` (which will be some kind of falsy value).

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 2]
```diff
@@ -121,10 +123,10 @@ const StartVisitForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, closeWor
         .pipe(first())
         .subscribe(
           (response) => {
+            getVisitFormData.currentVisit = response.data;
+            localStorage.setItem('visit-form', JSON.stringify(getVisitFormData));
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:127]: I've said this before, but please don't use `localStorage` for tracking temporary data. We really don't need any storage here and could likely just set a variable value, but if we _must_ use storage for this, use sessionStorage (local storage is for values that persist across sessions—not at all what we want here)

**Review by ibacher — COMMENTED** [review]
I get that this works, but it's very form specific. We have an overarching probably that whenever a workspace opens that requires a visit and we show the Start a Visit form, the user is not redirected to the workspace once they've started the visit, so it would be nicer if there were a way to solve this that wasn't specific to each thing in the workspace, i.e., if we can add the hooks to the part of the code that is actually opening the "Start Visit" workspace so that once that's complete, the expected workspace is opened.

File: packages/esm-patient-chart-app/src/actions-buttons/start-visit.component.tsx (hunk 1 above)
Comment (sainingo) [reply packages/esm-patient-chart-app/src/actions-buttons/start-visit.component.tsx:29]: Yes,.. I will revert that

**Pull request closed without merging**